pkg/asset/agent/image/unconfigured_ignition.go

```diff
@@ -208,6 +249,19 @@ func (a *UnconfiguredIgnition) Generate(_ context.Context, dependencies asset.Pa
 	return nil
 }
 
+func (a *UnconfiguredIgnition) ignAddFolders(config *igntypes.Config, paths ...string) {
+	for _, p := range paths {
+		config.Storage.Directories = append(config.Storage.Directories, igntypes.Directory{
+			Node: igntypes.Node{
+				Path:      p,
+				User:      igntypes.NodeUser{Name: ignutil.StrToPtr("root")},
+				Overwrite: ignutil.BoolToPtr(true),
+			},
+			DirectoryEmbedded1: igntypes.DirectoryEmbedded1{Mode: ignutil.IntToPtr(0644)},
+		})
+	}
+}
```

Comment:
Can you explain what this is for?

Reply:
Initially assisted-service.service, agent-import-cluster.service and agent-register-cluster.service required them
a.ignAddFolders(&config, "/opt/agent/tls", "/etc/assisted/extra-manifests"). Since, as per the latest discussions, the *cluster.service will not be triggered anymore, then only the `/opt/agent/tls` will be required.